js_modules/dagster-ui/packages/ui-core/src/util/dayjsExtensions.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable no-restricted-imports */
+import dayjs from 'dayjs';
+import duration from 'dayjs/plugin/duration';
+import relativeTime from 'dayjs/plugin/relativeTime';
+import timezone from 'dayjs/plugin/timezone';
+import updateLocale from 'dayjs/plugin/updateLocale';
+import utc from 'dayjs/plugin/utc';
+
+type ThresholdUnit = 'second' | 'minute' | 'hour' | 'day' | 'month' | 'year';
+
+const thresholds: {l: string; r?: number; d?: ThresholdUnit}[] = [
+  // Read as: "Format string to use when the diff (in seconds) <= r (44)"
+  // Followed by other format strings using the same diff until the next d:.
+  {l: 's', r: 44, d: 'second'},
+  {l: 'm', r: 89},
+  {l: 'mm', r: 44, d: 'minute'},
+  {l: 'h', r: 89},
+  {l: 'hh', r: 47, d: 'hour'}, // We have changed this from 21 to 47
+  //   {l: 'd', r: 35},           // We never show "A day ago"
+  {l: 'dd', r: 25, d: 'day'},
+  {l: 'M', r: 45},
+  {l: 'MM', r: 10, d: 'month'},
+  {l: 'y', r: 17},
+  {l: 'yy', d: 'year'},
+];
+
+dayjs.extend(duration);
+dayjs.extend(updateLocale);
+dayjs.extend(relativeTime, {thresholds});
+dayjs.extend(utc);
+dayjs.extend(timezone);
+
+dayjs.updateLocale('en', {
+  relativeTime: {
+    future: 'in %s',
+    past: '%s ago',
+    s: '%d seconds',
+    m: 'a minute',
+    mm: '%d minutes',
+    h: 'an hour',
+    hh: '%d hours',
+    d: 'a day',
+    dd: '%d days',
+    M: 'a month',
+    MM: '%d months',
+    y: 'a year',
+    yy: '%d years',
+  },
+});
+
+// Helper: returns ms until the relative time string will change for a given timestamp.
+// Used for timing the next update when displaying a relative time.
+//
+export function getNextFromNowUpdateMs(unixSeconds: number): number {
```

Comment:
Can we add some test cases for this

Reply:
Ahh no worries! This is complicated enough it really should have them -  you are not wrong :-p